File: 3. Longest Substring Without Repeating Characters/java/solution.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Set<Character> freq = new HashSet<>();
        int count = 0;
        int l = 0;
        int r = 0;
        while(r < s.length()){
            char c = s.charAt(r);
            if(freq.contains(c)){
                freq.remove(s.charAt(l));
                l++;
            }
            else{
                freq.add(c);
                count = Math.max(count, r - l + 1);
                r++;
            }
        }
        return count;
    }
}

File: 3. Longest Substring Without Repeating Characters/cpp/solution.cpp
class Solution {
    public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char, int> freq;
        int count = 0;
        int l = 0;
        int r = 0;
        while(r < s.size()){
            char c = s[r];
            freq[c]++;
            while (freq[c] > 1){
                freq[s[l]]--;
                l++;
            }
            count = max(count, r - l + 1);
            r++;
        }
        return count;
    }
};